archive/tar_unix.go

```diff
@@ -85,6 +85,7 @@ func openFile(name string, flag int, perm os.FileMode) (*os.File, error) {
 	}
 	// Call chmod to avoid permission mask
```

Comment:
Why would chmod fail when open succeeded with the same value of perm? While the change looks good from error handling perspective, i am curious to know what would cause chmod to fail here?

Reply:
The perm given to `OpenFile` would only be used when creating the file. This function could also be used to open existing files for modification. Seems unlikely but with tar archive every odd edge case has to be handled.

Reply:
From some experiments i did, it seems like, if it fails for existing file it will fail at open itself. But as i said before, having the extra check is definitely good to protect from corner cases. Thanks!